src/server/game/Entities/Creature/GossipDef.h

```diff
@@ -237,6 +234,21 @@ class TC_GAME_API QuestMenu
         QuestMenuItemList _questMenuItems;
 };
 
+class SelectionData
+{
+    public:
+        SelectionData() { Reset(); }
+
+        void Reset()
```

Comment:
Not sure what sends that or where/how it is used.
But if that is sent on gossip ending or closing windows like trainer window then it should reset most likely to prevent buying stuff etc after window has been closed - though I wouldnt consider that a big plus as you could prevent sending the packet through hacks anyways and then do the actions even after closing the window.

Seems that this system could also be applied to mails as Im looking at the comments in https://github.com/TrinityCore/TrinityCore/blob/b453e124231a90321fe79fbf3a62acdcfa54a691/src/server/game/Handlers/MailHandler.cpp#L385